index.js

```diff
@@ -1,64 +1,64 @@
 'use strict';
 
-var assign = require('object.assign');
+const { declare } = require('@babel/helper-plugin-utils');
 
-var modules = [require('babel-plugin-transform-es2015-modules-commonjs'), {
-  strictMode: true, // add "use strict"
-  strict: false // this allows __esModule to be exported
-}];
-
-var defaultTargets = {
+const defaultTargets = {
   android: 30,
   chrome: 35,
   edge: 14,
   explorer: 9,
   firefox: 52,
   safari: 8,
-  ucandroid: 1
+  ucandroid: 1,
 };
 
-function buildTargets(options) {
-  return assign({}, defaultTargets, options.additionalTargets);
+function buildTargets({ additionalTargets }) {
+  return Object.assign({}, defaultTargets, additionalTargets);
 }
 
-module.exports = function buildAirbnbPreset(context, options) {
-  var transpileTargets = (options && options.targets) || buildTargets(options || {});
+module.exports = declare((api, options) => {
+  api.assertVersion(7);
+
+  const {
+    targets = buildTargets(options),
+    modules,
+    removePropTypes,
+  } = options;
 
-  var debug = (options && typeof options.debug === 'boolean') ? !!options.debug : false;
+  const debug = typeof options.debug === 'boolean' ? options.debug : false;
+  const development = typeof options.development === 'boolean' ? options.development : process.env.NODE_ENV === 'development';
```

Comment:
Checks like `process.env.NODE_ENV === 'development'` should be wrapped for caching, so you can do
```
const development = typeof options.development === 'boolean' ? options.development : api.cache.using(() => process.env.NODE_ENV === 'development');
```
so that Babel knows to re-evaluate the preset if the `NODE_ENV` changes.

Reply:
Also if it helps, there's some docs for this API here: https://babeljs.io/docs/en/config-files#apicache

Reply:
Ok cool, I also couldn't find an explicit return type in those docs so I had to dig through [the source](https://github.com/babel/babel/blob/2a4f16236656178e84b05b8915aab9261c55782c/packages/babel-core/src/config/caching.js#L152).